[CI] QueueResizingEsThreadPoolExecutorTests testAutoQueueSizingWithMin failing

Similar failure happened with another test in this suite a couple weeks back: https://gradle-enterprise.elastic.co/s/ntpidbhmvfftc/tests/:server:test/org.elasticsearch.common.util.concurrent.QueueResizingEsThreadPoolExecutorTests/testAutoQueueSizingWithMax

Pretty infrequent though. Only 2 failures in 6000 executions but seems like there might be a way to make it more robust.

**Build scan:**
https://gradle-enterprise.elastic.co/s/bnqpedlpnwoik/tests/:server:test/org.elasticsearch.common.util.concurrent.QueueResizingEsThreadPoolExecutorTests/testAutoQueueSizingWithMin

**Reproduction line:**
`./gradlew ':server:test' --tests "org.elasticsearch.common.util.concurrent.QueueResizingEsThreadPoolExecutorTests.testAutoQueueSizingWithMin" -Dtests.seed=6D075E6DDC08AEEB -Dtests.security.manager=true -Dtests.locale=en-IN -Dtests.timezone=Asia/Urumqi -Druntime.java=8`

**Applicable branches:**
7.x

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.common.util.concurrent.QueueResizingEsThreadPoolExecutorTests&tests.test=testAutoQueueSizingWithMin

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <4989>
     but: was <5039>

  at __randomizedtesting.SeedInfo.seed([6D075E6DDC08AEEB:2EB7CF3FBEEA9555]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.common.util.concurrent.QueueResizingEsThreadPoolExecutorTests.lambda$testAutoQueueSizingWithMin$6(QueueResizingEsThreadPoolExecutorTests.java:136)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1019)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:992)
  at org.elasticsearch.common.util.concurrent.QueueResizingEsThreadPoolExecutorTests.testAutoQueueSizingWithMin(QueueResizingEsThreadPoolExecutorTests.java:135)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

Similar error for testAutoQueueSizingWith**Max**

https://gradle-enterprise.elastic.co/s/eddwan5oivtwo/tests/:server:test/org.elasticsearch.common.util.concurrent.QueueResizingEsThreadPoolExecutorTests/testAutoQueueSizingWithMax

```
java.lang.AssertionError: |  
Expected: <5015> |  
  | but: was <4965>

at __randomizedtesting.SeedInfo.seed([7B733D3A04F48362:E76062F422CE56D1]:0) |  
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
at org.junit.Assert.assertThat(Assert.java:956) |  
at org.junit.Assert.assertThat(Assert.java:923) |  
at org.elasticsearch.common.util.concurrent.QueueResizingEsThreadPoolExecutorTests.lambda$testAutoQueueSizingWithMax$7(QueueResizingEsThreadPoolExecutorTests.java:165)
```

--- a/server/src/test/java/org/elasticsearch/common/util/concurrent/QueueResizingEsThreadPoolExecutorTests.java
+++ b/server/src/test/java/org/elasticsearch/common/util/concurrent/QueueResizingEsThreadPoolExecutorTests.java
@@ -110,64 +110,6 @@ public class QueueResizingEsThreadPoolExecutorTests extends ESTestCase {
         executor.awaitTermination(10, TimeUnit.SECONDS);
     }
 
-    public void testAutoQueueSizingWithMin() throws Exception {
-        ThreadContext context = new ThreadContext(Settings.EMPTY);
-        ResizableBlockingQueue<Runnable> queue =
-                new ResizableBlockingQueue<>(ConcurrentCollections.<Runnable>newBlockingQueue(),
-                        5000);
-
-        int threads = randomIntBetween(1, 5);
-        int measureWindow = randomIntBetween(10, 100);
-        int min = randomIntBetween(4981, 4999);
-        logger.info("--> auto-queue with a measurement window of {} tasks", measureWindow);
-        QueueResizingEsThreadPoolExecutor executor =
-                new QueueResizingEsThreadPoolExecutor(
-                        "test-threadpool", threads, threads, 1000,
-                        TimeUnit.MILLISECONDS, queue, min, 100000, slowWrapper(), measureWindow, TimeValue.timeValueMillis(1),
-                        EsExecutors.daemonThreadFactory("queuetest"), new EsAbortPolicy(), context);
-        executor.prestartAllCoreThreads();
-        logger.info("--> executor: {}", executor);
-
-        // Execute a task multiple times that takes 1m
-        executeTask(executor, (measureWindow * 5));
-
-        // The queue capacity should decrease, but no lower than the minimum
-        assertBusy(() -> {
-            assertThat(queue.capacity(), equalTo(min));
-        });
-        executor.shutdown();
-        executor.awaitTermination(10, TimeUnit.SECONDS);
-    }
-
-    public void testAutoQueueSizingWithMax() throws Exception {
-        ThreadContext context = new ThreadContext(Settings.EMPTY);
-        ResizableBlockingQueue<Runnable> queue =
-                new ResizableBlockingQueue<>(ConcurrentCollections.<Runnable>newBlockingQueue(),
-                        5000);
-
-        int threads = randomIntBetween(1, 5);
-        int measureWindow = randomIntBetween(10, 100);
-        int max = randomIntBetween(5010, 5024);
-        logger.info("--> auto-queue with a measurement window of {} tasks", measureWindow);
-        QueueResizingEsThreadPoolExecutor executor =
-                new QueueResizingEsThreadPoolExecutor(
-                        "test-threadpool", threads, threads, 1000,
-                        TimeUnit.MILLISECONDS, queue, 10, max, fastWrapper(), measureWindow, TimeValue.timeValueMillis(1),
-                        EsExecutors.daemonThreadFactory("queuetest"), new EsAbortPolicy(), context);
-        executor.prestartAllCoreThreads();
-        logger.info("--> executor: {}", executor);
-
-        // Execute a task multiple times that takes 1ms
-        executeTask(executor, measureWindow * 3);
-
-        // The queue capacity should increase, but no higher than the maximum
-        assertBusy(() -> {
-            assertThat(queue.capacity(), equalTo(max));
-        });
-        executor.shutdown();
-        executor.awaitTermination(10, TimeUnit.SECONDS);
-    }
-
     public void testExecutionEWMACalculation() throws Exception {
         ThreadContext context = new ThreadContext(Settings.EMPTY);
         ResizableBlockingQueue<Runnable> queue =

